### PR TITLE
Makes imaginary port configurable

### PR DESCRIPTION
**Pull request by steffenmalisi**
With commit https://github.com/nextcloud/all-in-one/commit/6fe6d7e0b4da44a57b1f1c1226b40fb712b50a22 the imaginary image lost the possibility of manually configuring the port imaginary listens to.

The original image, where this one was based on (https://github.com/h2non/imaginary/blob/master/Dockerfile), does support defining the port as ENV variable.

This PR introduces this capability again, also for AIO.

Why is it needed? In my case I am running some of the AIO containers in one Podman pod. Within a pod all containers share the same network namespace. Because php fpm and imaginary are using port 9000 by default, I welcomed the possibility of the imaginary image to change the default port.

### CODE REVIEW

File: Containers/imaginary/Dockerfile [hunk 1]
```diff
@@ -23,11 +23,13 @@ RUN set -ex; \
...
+
 USER nobody
 
 # https://github.com/h2non/imaginary#memory-issues
 ENV MALLOC_ARENA_MAX=2
-ENTRYPOINT ["imaginary", "-p", "9000", "-return-size", "-max-allowed-resolution", "222.2"]
+ENTRYPOINT ["imaginary", "-return-size", "-max-allowed-resolution", "222.2"]
 
-HEALTHCHECK CMD nc -z localhost 9000 || exit 1
+HEALTHCHECK CMD nc -z localhost $PORT || exit 1
```
Comment (Zoey2936) [inline Containers/imaginary/Dockerfile:34]:
```suggestion
HEALTHCHECK CMD nc -z localhost "$PORT" || exit 1

```
maybe add ""?

**Review by Zoey2936 — APPROVED** [review]

File: Containers/imaginary/Dockerfile [hunk 2]
```diff
@@ -7,7 +7,7 @@ RUN set -ex; \
         vips-jxl \
         vips-poppler \
         build-base; \
-    go install github.com/h2non/imaginary@b632dae8cc321452c3f85bcae79c580b1ae1ed84
+    go install github.com/h2non/imaginary@b632dae8cc321452c3f85bcae79c580b1ae1ed84
```
Comment (szaimen) [inline Containers/imaginary/Dockerfile:10]: This looks suspicious

Comment (steffenmalisi) [reply Containers/imaginary/Dockerfile:10]: Nothing suspicious in here.

The difference is the carriage return (^M) at the end of the line. Unfortunately you can not see this within Github, but you can, if you do a `git diff`

image

Where does it come from? This is what I have configured: https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings

Comment (szaimen) [reply Containers/imaginary/Dockerfile:10]: Can you remove the ^M again? Then I can merge

Comment (steffenmalisi) [reply Containers/imaginary/Dockerfile:10]: I can, if you really want.

But first let me try again to explain, why I think it's more consistent to leave it in.

`cat -A` shows the invisible carriage return characters. Now look at the respective line before the PR. This line is the only one not including ^M (equals to '\r' or 'CR'). ^M$ means \r\n or CRLF, which can be interpreted by all platforms and therefore is preferable when multiple people collaborate.

Pre PR :
```
cat -A Containers/imaginary/Dockerfile

FROM golang:1.20.4-alpine3.17 as go^M$
RUN set -ex; \^M$
    apk add --no-cache \^M$
        vips-dev \^M$
        vips-magick \^M$
        vips-heif \^M$
        vips-jxl \^M$
        vips-poppler \^M$
        build-base; \^M$
    go install github.com/h2non/imaginary@b632dae8cc321452c3f85bcae79c580b1ae1ed84$
^M$
FROM alpine:3.17.3^M$
RUN set -ex; \^M$
    apk add --no-cache \^M$
        tzdata \^M$
        ca-certificates \^M$
        netcat-openbsd \^M$
        vips \^M$
        vips-magick \^M$
        vips-heif \^M$
        vips-jxl \^M$
        vips-poppler^M$
^M$
COPY --from=go /go/bin/imaginary /usr/local/bin/imaginary^M$
^M$
USER nobody^M$
^M$
# https://github.com/h2non/imaginary#memory-issues^M$
ENV MALLOC_ARENA_MAX=2^M$
ENTRYPOINT ["imaginary", "-p", "9000", "-return-size", "-max-allowed-resolution", "222.2"]^M$
^M$
HEALTHCHECK CMD nc -z localhost 9000 || exit 1^M$
LABEL com.centurylinklabs.watchtower.monitor-only="true"^M$
```

After PR

```
FROM golang:1.20.4-alpine3.17 as go^M$
RUN set -ex; \^M$
    apk add --no-cache \^M$
        vips-dev \^M$
        vips-magick \^M$
        vips-heif \^M$
        vips-jxl \^M$
        vips-poppler \^M$
        build-base; \^M$
    go install github.com/h2non/imaginary@b632dae8cc321452c3f85bcae79c580b1ae1ed84^M$
^M$
FROM alpine:3.17.3^M$
RUN set -ex; \^M$
    apk add --no-cache \^M$
        tzdata \^M$
        ca-certificates \^M$
        netcat-openbsd \^M$
        vips \^M$
        vips-magick \^M$
        vips-heif \^M$
        vips-jxl \^M$
        vips-poppler^M$
^M$
COPY --from=go /go/bin/imaginary /usr/local/bin/imaginary^M$
^M$
ENV PORT 9000^M$
^M$
USER nobody^M$
^M$
# https://github.com/h2non/imaginary#memory-issues^M$
ENV MALLOC_ARENA_MAX=2^M$
ENTRYPOINT ["imaginary", "-return-size", "-max-allowed-resolution", "222.2"]^M$
^M$
HEALTHCHECK CMD nc -z localhost "$PORT" || exit 1^M$
LABEL com.centurylinklabs.watchtower.monitor-only="true"^M$
```

Hopefully it's now a bit clearer. But if you want me to still use the previous version, just give me a hint.

Comment (szaimen) [reply Containers/imaginary/Dockerfile]: you have a point here. I think I fixed the problem in main and rebased the branch. Ready to go now

**Review by szaimen — APPROVED** [review]

**Comment by szaimen** [comment]
This is now released with v5.2.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Comment by steffenmalisi** [comment]
> This is now released with v5.2.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

Works in my setup
